node/task.ts

```diff
@@ -738,16 +738,30 @@ export function ls(options: string, paths: string[]): string[] {
  * @param     dest       destination path
  * @param     options    string -r, -f or -rf for recursive and force
  * @param     continueOnError optional. whether to continue on error
+ * @param     retryCount optional. Retry count to copy the file
  */
-export function cp(source: string, dest: string, options?: string, continueOnError?: boolean): void {
+export function cp(source: string, dest: string, options?: string, continueOnError?: boolean, retryCount: number = 0): void {
     if (options) {
         shell.cp(options, source, dest);
     }
     else {
         shell.cp(source, dest);
     }
 
-    _checkShell('cp', continueOnError);
+    try {
+        _checkShell('cp', false);
+    } catch (e) {
+        if (retryCount > 0) {
+            console.log(loc('LIB_CopyFileFailed', retryCount));
+            cp(source, dest, options, continueOnError, retryCount - 1);
```

Comment:
Nit: could it make sense to make it iterative? This would make this code more robust from the point of reaching stack limits

Reply:
Thanks for the point. Reworked it.